### PR TITLE
Add a positions object to output quadrature points

### PR DESCRIPTION
**Pull request by GiudGiud**
refs https://github.com/idaholab/moose/issues/23587

### CODE REVIEW

File: framework/src/positions/QuadraturePointsPositions.C [hunk 1]
```diff
@@ -0,0 +1,92 @@
...
+    unsigned int b_index = 0;
+    for (const auto & sub_id : blockIDs())
+    {
+      for (const auto & elem : _mesh.getMesh().active_local_subdomain_elements_ptr_range(sub_id))
+      {
+        // Get a quadrature going on the element
+        const FEFamily mapping_family = FEMap::map_fe_type(*elem);
+        const FEType fe_type(elem->default_order(), mapping_family);
+        std::unique_ptr<FEBase> fe = FEBase::build(elem->dim(), fe_type);
+        QGauss qrule(elem->dim(), fe_type.default_quadrature_order());
```
Comment (lindsayad) [inline framework/src/positions/QuadraturePointsPositions.C:59]: I feel like a user would more naturally expect this to match what’s used in Assembly which is controllable using the input file where as this hard coding is not

Comment (GiudGiud) [reply framework/src/positions/QuadraturePointsPositions.C]: ok changed. matching assembly comes at a cost (cant run as early, add_positions task is very early) though so I did not do exactly that

**Review by lindsayad — APPROVED** [review]
the change works for me

**Comment by GiudGiud** [comment]
Thanks for the review!

**Pull request merged**